replace --forward-port directions with --add-port in the port-forwarding page

yyolk opened the pull request:
- replacing `--forward-port` directions with `--add-port`
  - readability add note for NAT and routing (thanks @mDuo13 !)
- add note for virtual firewall and attaching 

I'm new to firewalld, but I think the purpose of the [`--add-forward-port`](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/sec-port_forwarding) is if it's served internally on a different port. 


Update: removed a bunch of cruft around being a bit of a novice in running a node. I've now successfully done it on bare metal (requiring a firewall), and on various VPS several times including.

I've run a test node on 'bare metal' for a month, which I opted in using `firewalld` to the internet.

I started a fresh one 5 days ago using the latest commit's [changes as detailed in this post](https://github.com/ripple/xrpl-dev-portal/pull/1105#issuecomment-873455189) (caught my wrong long option :sweat_smile:)

yyolk committed:
replace --forward-port directions with --add-port

replacing --forward-port directions with --add-port

I'm new to firewalld, but I think the purpose of the [`--add-forward-port`](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/sec-port_forwarding) is if it's served internally on a different port.

It very well may work as an open-port, maybe if there's a bridged network? (i.e., `docker`); I didn't have luck with it. I came to the page to see what port I needed to open with the command I'm presenting, but tried this approach first. I didn't get any peer connections. When I switched over to just opening the port as `51235/tcp` I quickly saw peers.

I am also new to running a node, and this particular one was running for around 2 weeks without peering which may have caused some difference.

I do also think this file could remain the same name as it's capturing it's intent.

Co-authored-by: Rome Reginelli <mduo13@gmail.com>

typo

Update content/tutorials/manage-the-rippled-server/configure-peering/forward-ports-for-peering.md

Update content/tutorials/manage-the-rippled-server/configure-peering/forward-ports-for-peering.md

`--add-port` not `--add-source-port`

add note for virtual firewall

Co-authored-by: Rome Reginelli <mduo13@gmail.com>

SirJosh1987 reviewed (approved):
Am I on the right path so far

mDuo13 commented on content/tutorials/manage-the-rippled-server/configure-peering/forward-ports-for-peering.md:
````diff
@@ -35,10 +35,24 @@ Loading: "/etc/opt/ripple/rippled.cfg"
+$ sudo firewall-cmd --zone=public --add-source-port=51235/tcp
+```
+
+It's a good idea to restart rippled to see it's effects:
````
```suggestion
Then, restart the `rippled` server:
```

mDuo13 commented on content/tutorials/manage-the-rippled-server/configure-peering/forward-ports-for-peering.md:
````diff
@@ -35,10 +35,24 @@ Loading: "/etc/opt/ripple/rippled.cfg"
 ```
 
-To allow incoming connections, configure your firewall to forward the peer protocol port, which is served on **port 51235** in the default config file. The instructions to forward a port depend on your firewall. For example, if you use the `firewalld` software firewall on Red Hat Enterprise Linux, you can [use the `firewall-cmd` tool](https://access.redhat.com/documentation/en-us/red_hat_enterpri ...
+To allow incoming connections, configure your firewall to forward the peer protocol port, which is served on **port 51235** in the default config file. The instructions to forward a port depend on your firewall. For example, if you use the `firewalld` software firewall on Red Hat Enterprise Linux, you can [use the `firewall-cmd` tool](https://access.redhat.com/documentation/en-us/red_hat_enterpri ...
````
```suggestion
To allow incoming connections, configure your firewall to allow incoming traffic on the peer protocol port, which is served on **port 51235** in the default config file. The instructions to open a port depend on your firewall. If your server is behind a router that performs Network Address Translation (NAT), you must configure your router to forward the port to your server.

If you use the `firewalld` software firewall on Red Hat Enterprise Linux, you can [use the `firewall-cmd` tool](https://access.redhat.com/documentation/en-us/red_hat_enterprise_linux/7/html/security_guide/sec-using_zones_to_manage_incoming_traffic_depending_on_source) to open **port 51235** to all incoming traffic.
```

yyolk replied:
Great suggestion!

SirJosh1987 replied:
How?

mDuo13 reviewed (changes requested):
- Rephrased the "restart" bit. (You had the wrong "its" there)
- While we're here, a misconception I've seen a couple times is that people think they're not behind a firewall even though their NAT router is acting as a firewall. So I suggested a rephrased first paragraph to clarify that.
- I'm not familiar enough with `firewalld` to confirm that your instructions are right, but I think they might be. I'll see if I can get someone else with more expertise to double-check.

yyolk commented:
>     * Rephrased the "restart" bit. (You had the wrong "its" there)
Thank you :)

>     * While we're here, a misconception I've seen a couple times is that people think they're not behind a firewall even though their NAT router is acting as a firewall. So I suggested a rephrased first paragraph to clarify that.

Totally makes sense! I encountered this page looking for _"the port to forward"_ having the NAT bit is useful context. :+1: 

 
>     * I'm not familiar enough with `firewalld` to confirm that your instructions are right, but I think they might be. I'll see if I can get someone else with more expertise to double-check.

I believe this as well. I'm far from an expert with `firewalld`, I did try the previous `--add-forward-port` on a bare metal server with `firewalld` and didn't get anywhere. I'll run through this again to confirm.

yyolk commented:
here's documentation running through ~~`--add-forward-port`~~ `--add-port` (*[see checklist item](#lolwhoops) where i added commit https://github.com/ripple/xrpl-dev-portal/pull/1105/commits/54e6ea2866220245af196a5ff2d81a6ea4d2201a*)

starting from a new instance, no rules are for forward or open ports:

![image](https://user-images.githubusercontent.com/134478/124363484-175d4a80-dc01-11eb-86ba-4dbe7fd71ec9.png)

adding the forward-port
![image](https://user-images.githubusercontent.com/134478/124363797-031a4d00-dc03-11eb-9718-1eac64feea5b.png)


initial sync, seems that more peers are being seen that anticipated (i recall seeing 10 here, but this was before i started to set `[peers_max]`)
![image](https://user-images.githubusercontent.com/134478/124363569-a10d1800-dc01-11eb-92c4-d3a016ab6139.png)

still not "FULL", this node is connecting to peers but nothing is able to connect to it:

![image](https://user-images.githubusercontent.com/134478/124363664-37413e00-dc02-11eb-9163-455245692fc4.png)

node is "FULL", no peers are connecting to it:
- ![image](https://user-images.githubusercontent.com/134478/124363684-5fc93800-dc02-11eb-9821-7fd98f4dbe86.png)
- ![image](https://user-images.githubusercontent.com/134478/124363692-6fe11780-dc02-11eb-858a-215b0178e466.png)


* * *
stopped `rippled` and removed forward rule and add open port rule

![image](https://user-images.githubusercontent.com/134478/124363850-7cb23b00-dc03-11eb-9e52-f3c225e25240.png)

* * *

starting up and waiting no time, there are no peers:

![image](https://user-images.githubusercontent.com/134478/124363961-0a8e2600-dc04-11eb-8569-96fe577be6ca.png)


so i deleted the peerfinder sqllite db in `/var/lib/rippled/db/`

***turns out i actually need to use `--add-port` (which matches my production-test node, not this new test) :sweat_smile:***

![image](https://user-images.githubusercontent.com/134478/124364563-e16f9480-dc07-11eb-87d7-e5e359f60c9a.png)


- [X] <a name="lolwhoops"></a> fix to use `--add-port`, not `--add-source-port`


immediately i see nodes connect to me with it
![image](https://user-images.githubusercontent.com/134478/124364035-7a041580-dc04-11eb-80d1-985c328a387d.png)
![image](https://user-images.githubusercontent.com/134478/124364096-e54de780-dc04-11eb-95c9-c166c28ed348.png)

finally an established connection:
![image](https://user-images.githubusercontent.com/134478/124364138-20e8b180-dc05-11eb-9827-08256f3c1ddc.png)

waited a little bit, now i'm seeing peers on the node list
![image](https://user-images.githubusercontent.com/134478/124364397-e253f680-dc06-11eb-9f46-0462f074dfa6.png)

yyolk commented:
added a note around virtual firewalls, updated my pull description

mDuo13 commented on content/tutorials/manage-the-rippled-server/configure-peering/forward-ports-for-peering.md:
```diff
@@ -35,14 +35,32 @@ Loading: "/etc/opt/ripple/rippled.cfg"
 
 For other software and hardware firewalls, see the manufacturer's official documentation.
 
+If your using a hosting provider includes a virtual firewall, you do not need to use `firewalld`, but will still need to allow the peer port, and ensure your host is attached. e.g., AWS Security Groups with a rule for the peer port open to the public internet.
```
```suggestion
If you are using a hosting service with a virtual firewall (for example, [AWS Security Groups](https://docs.aws.amazon.com/vpc/latest/userguide/VPC_SecurityGroups.html)), you do not need to use `firewalld`, but you still need to allow inbound traffic from the open internet on the peer port. Make sure you apply the relevant rules to your host or virtual machine.
```

mDuo13 reviewed (changes requested):
Looks like "your hosting provider" and "you're using a hosting provider" got mixed up into a typo here. I suggested a rephrased version to clarify. (Not sure what "ensure your host is attached" is supposed to have meant there—I took a guess that you mean to remember to attach the rules to the relevant machine.)

SirJosh1987 commented:
I'm slowly learning

mDuo13 committed:
Update port forwarding (rephrase around typo)

yyolk commented:
Thank you!
